src/components/field-label/field-label.component.tsx

```diff
@@ -13,19 +14,40 @@ interface FieldLabelProps {
   customLabel?: string;
 }
 
+const TooltipWrapper: React.FC<{ hasTooltip: boolean; field: FormField; children: React.ReactNode }> = ({
+  field,
+  children,
+  hasTooltip,
+}) => {
+  const { t } = useTranslation();
+  return hasTooltip ? (
+    <CarbonTooltip align="top-left" label={t(field.questionInfo)}>
+      {children}
+    </CarbonTooltip>
+  ) : (
+    <>{children}</>
+  );
+};
+
 const FieldLabel: React.FC<FieldLabelProps> = ({ field, customLabel }) => {
   const { t } = useTranslation();
+  const hasTooltip = Boolean(field.questionInfo);
   const labelText = customLabel || t(field.label);
+
   return (
-    <div className={styles.questionLabel}>
-      <span>{labelText}</span>
-      {field.isRequired && (
-        <span title={t('required', 'Required')} className={styles.required}>
-          *
-        </span>
-      )}
-      {field.questionInfo && <Tooltip field={field} />}
-    </div>
+    <TooltipWrapper field={field} hasTooltip={hasTooltip}>
+      <div className={styles.questionLabel} data-testid={`${field.id}-label`}>
+        <span>{labelText}</span>
+        {field.isRequired && (
+          <span title={t('required', 'Required')} className={styles.required}>
+            *
+          </span>
+        )}
+        {hasTooltip && (
+          <Information size={20} aria-hidden="true" className={styles.tooltipIcon} data-testid="information-icon" />
```

Comment:
@vasharma05 do you think we should include the field's ID in the `data-testid` to guarantee its uniqueness?